packages/office-ui-fabric-react/src/components/ExtendedPicker/PeoplePicker/ExtendedPeoplePicker.tsx

```diff
@@ -1,17 +1,26 @@
 /* tslint:disable */
-import { IPickerItemProps } from '../../../Pickers';
+import { IPickerItemProps, IInputProps } from '../../../Pickers';
 /* tslint:enable */
 
 import { IExtendedPersonaProps } from '../../../SelectedItemsList';
 import { IPersonaProps } from '../../../Persona';
 import './ExtendedPeoplePicker.scss';
 import { BaseExtendedPicker } from '../BaseExtendedPicker';
 import { IBaseExtendedPickerProps } from '../BaseExtendedPicker.types';
+import { assign } from '@uifabric/utilities';
 
 export interface IPeoplePickerItemProps extends IPickerItemProps<IExtendedPersonaProps> {}
 
 export interface IExtendedPeoplePickerProps extends IBaseExtendedPickerProps<IPersonaProps> {}
 
 export class BaseExtendedPeoplePicker extends BaseExtendedPicker<IPersonaProps, IExtendedPeoplePickerProps> {}
 
-export class ExtendedPeoplePicker extends BaseExtendedPeoplePicker {}
+export class ExtendedPeoplePicker extends BaseExtendedPeoplePicker {
```

Comment:
I don't think that this is the right way to achieve this. Instead you could do something like 
```
export class ExtendedPeoplePicker extends BaseExtendedPeoplePicker {
  protected _skipComponentRefResolution = true;
  public render() {
    return <BaseExtendedPeoplePicker 
    {...this.props} 
    inputProps = {{...this.props.inputProps, ...overwrittenInputProps}}
    />
  }
}

Reply:
@JasonGore may have some thoughts on how to _best_ override component props from the base class here when he has a moment.

Reply:
The best way (and officia React way) is [not to use inheritance at all](https://github.com/facebook/react/issues/4806#issuecomment-378284255).

In this case, I think the sanest approach is just to do:

```tsx
class ExtendedPeoplePicker extends React.Component {
    render() {
        return <BaseExtendedPeoplePicker {...this.props} inputProps={{...this.props.inputProps, ...overwrittenInputProps} />;
    }
}
```

If there is a reason not to use composition, then at the very least @joschect's advice should be taken as it doesn't require the base class to be modified at all and `inputProps` member can be removed.

Reply:
(Changed to `extends BaseComponent` to make sure refs still work.)

Reply:
Thanks for the input @joschect and @JasonGore! My initial thought was to go with something similar to what @joschect suggested, but I thought it was a bit weird to render the component that it's also extending. I was aware of the official recommendation of not using component inheritance, but I didn't want to make that assumption initially without further input.

I like @JasonGore's suggestion, but am happy to go with @joschect if desired.

Reply:
Oops, I was wrong about my comment. Extending `React.Component` should be sufficient. Similar to our other HOC wrappers (styled, etc.) the refs will just get forwarded in the props and should continue to work fine.

Reply:
@aleury Go with Jason's suggestion, it looks more correct.